Dockerfile

```diff
@@ -1,30 +1,71 @@
-FROM alpine:3.6
+FROM alpine:3.8
 
-ARG FROM_REPOSITORY=https://github.com/calzoneman/sync
+ENV MYSQL_HOST=localhost \
+	MYSQL_PORT=3306 \
+	MYSQL_DATABASE=cytube3 \
+	MYSQL_USER=cytube3 \
+	MYSQL_PASSWORD=UltraSecretPass \
+	MYSQL_ROOT_PASSWORD=UltraSecretRootPass \
+	HTTP_PORT=8080 \
+	HTTP=true \
+	HTTPS_PORT=8443	\
+	HTTPS=false \
+	IO=true \
+	IO_PORT=1337 \
+	IO_DOMAIN=http://localhost \
+	ROOT_DOMAIN=localhost \
+	USE_MINIFY=false \
+	COOKIE_SECRET=change-me \
+	SYNC_CRTKEY=null \
+	SYNC_CRT=null \
+	SYNC_CRTCA=null \
+	SYNC_TITLE=Sync \
+	SYNC_DESCRIPTION="Free, open source synchtube" \
+	YOUTUBE_KEY=null \
+	CHANNEL_STORAGE=file \
+	VIMEO_WORKAROUND=false \
```

Comment:
For example, I don't think this config switch should be needed by anyone but me (it predates the time when I started maintaining a separate fork with cytu.be-specific patches)

Reply:
I left that in with you in mind, but since it's set to the default value it isn't any different than the mainline config template; shouldn't be any harm having it included but I do agree that it can be confusing to the uninitiated.

Reply:
Regarding your comment about synchronizing configs and other attributes, you should look into an NFS-based solution. Another good option would be an orchestration tool to automate manual syncs. Quick example with this dockerfile; you could set this up as a docker machine, docker swarm, kubernetes or rancher cluster, whatever, map an NFS dir outside of the container system to any directories you want persistent and synced, and then use such tools to automatically deploy additional instances and load balancing when necessary. You could have a single running instance that you pay for by usage and have it scale as needed. That's just another use case of this project that may be worth looking into.

Regarding making it more difficult with maintenance in the future, I don't disagree, but I see that as a future goal along with the mainline cytube repo. This container in its current state accurately represents the build and deployment procedure in the docs as well as the config templates that current exist in the mainline repo. I agree that it can and should be cleaned up in the future, but I don't think that's a reason to delay the dockerfile in its current state. It doesn't make a difference to me; I'm using it regardless. I just don't see any harm in making this more readily available at least for the sake of development. The easiest way to get people to help out with testing and debugging would be to provide a stable development image, which is ultimately what this provides by default. Additionally the ease of deployment will make the project more attractive to newcomers and perhaps lead to a surplus in active developers or users (which is helpful for feedback). Not to mention it's use as a utility when considering infrastructure upgrades like I suggested above.

In regard to the config file being used for diagnosis, a script could always be made that parses the config into a something easier to digest for sharing. That being said, that should be in the mainline repo and not in the docker image itself. Regardless, when you deploy this image, the run.sh script outputs a config.yaml with all env vars replaced with their currently-set values, so this deployment method does not get in the way of users being able to send you a config.yaml file.

If you haven't attempted to run this yet I would suggest doing so. I think you may have a slightly different perspective after witnessing the deployment process in its current state. I've tried it through CLI on Windows and Linux hosts, I've used it with the GUI tools Kitematic and Dockstation using the official Docker for Windows local hosting solution, as well as on a Linux host through Portainer. For anyone satisfied with the default variables (which are suitable for creating a test/dev instance), this is extremely easy to roll out. This can be deployed without any additional configuration on every up-to-date management GUI available using "gelmo/cytube-docker:fresh" as the image, and via CLI on any Docker host with this command:

docker run -ti gelmo/cytube-docker:fresh

That results in a fresh, working cytube instance on localhost with STDOUT in the foreground. This is the easiest way to spin up a fresh instance based on changes you want to push to the repo or from another branch, while being able to view the output in realtime. I agree that more work should be done to make this friendlier for production usage, but I think once you've seen it in action you'll agree that the current state is suitable for release, so long as the README indicates that it is currently not designed for production out of the box. Any changes that aren't related to making a production instance are changes which need to be made in the cytube repo first. I can't design this for a configuration system that doesn't exist, but I can say that the current design works for the configuration system which is currently being used.

I'm really sorry for being so difficult; I just want to make sure we're on the same page

Reply:
Additionally, I am fully committing to maintaining this container and whatever it eventually evolves into. I am moving my entire infrastructure (including sites that do not use cytube) to containers, and will be using orchestration tools to facilitate deployment, maintenance, monitoring, etc. This container will continue to improve over time to make it easier for such a use case; I intend on using cytube for the unforeseeable future and this will be how I do so. I am more than happy to help out with any issues that are related to deployment with Docker, whether it's in this repo or the mainline.

Reply:
If it wasn't clear, the specific examples of configuration synchronization and people pasting me config files were not intended to be used as arguments against docker, rather, I wanted to provide some examples to illustrate and provide context for my argument that the current CyTube configuration system is flawed and needs work.

I think the difference in opinion is rooted in a different perspective on what the docker repo is.  As a maintainer of the project for 5 years, I'm biased to see it as a fresh opportunity to create a new, alternative installation/configuration mechanism that doesn't carry the baggage of years of backwards compatibility and legacy features for existing installations.  As a server admin looking to migrate their infrastructure to containers, I think you see it more as a reification or perhaps a convenience of the existing installation method, and in that viewpoint configuration parity makes sense.

There's no need to apologize for engaging in discussion, at least it shows you have some vision for the project and aren't just trying to drive-by a pull request.